src/measurement.rs

```diff
@@ -0,0 +1,157 @@
+use bitfield::{bitfield, Bit, BitRange};
+
+#[derive(Copy, Clone, Default)]
+pub struct U12(u16);
```

Comment:
You need to expose the inner type, otherwise it can't be constructed:

```suggestion
pub struct U12(pub u16);
```

Alternatively you could create a constructor with an assertion that the value fits into 12 bits. Not sure if that wouldn't be unnecessary overhead though (and we'd probably prefer obviously broken data over a crash).

Reply:
We could also saturate in the constructor: `value.max(0xFFF)`

Reply:
I added a saturating constructor.